apigw-lambda-comprehend-language-sam-nodejs/template.yaml

```diff
@@ -0,0 +1,41 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: A REST API is created with lambda back end integration that detects the dominant language in the provided input text using Amazon Comprehend.
+Resources:
+  DetectLanguageApi:
+    Type: AWS::Serverless::Api
+    Properties:
+      StageName: dev
+      OpenApiVersion: '2.0'
+
+  DetectLanguageLambdaFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: index.handler
+      Runtime: nodejs20.x
+      CodeUri: src/
+      MemorySize: 128
+      Architectures:
+        - arm64
+      Timeout: 10
+      Policies:
+        - Version: '2012-10-17'
+          Statement:
+          - Sid: LambdaDetectDominantLanguage
+            Effect: Allow
+            Action:
+            - comprehend:DetectDominantLanguage
+            Resource: '*'
```

Comment:
Hey @abilashkumar same as the other pattern I reviewed, is the * resource required here? Can we lock this down at all?

Reply:
@boyney123 This is required as the permission here is not tied to a resource, but in general to perform language detection on free text provided as an input. The "*" is required (it does not involve any customer resource or their saved configurations). A similar pattern published [earlier](https://github.com/aws-samples/serverless-patterns/blob/main/lambda-translator/template.yaml#L20).